lib/client.js

```diff
@@ -435,12 +439,23 @@ class Parser extends HTTPParser {
       return true
     }
 
+    if (headers['keep-alive']) {
+      const m = headers['keep-alive'].match(/timeout=(\d+)/)
+      if (m) {
+        const keepAliveTimeout = parseInt(m[1]) * 1000
+        // Set timeout to half of hint to account for timing inaccuracies.
+        client[kKeepAliveTimeout] = Math.min(keepAliveTimeout / 2, client[kIdleTimeout])
+      }
+    } else {
+      client[kKeepAliveTimeout] = client[kIdleTimeout]
```

Comment:
Is this line necessary?

Reply:
I think so. If the server does not provide a hint for subsequent requests I think we should fallback to the idle timeout option.

Reply:
But you already set this at line 137, no?

Reply:
Not sure I follow. Line 447 will override line 137.

Reply:
I mean line 450 is the same as 137.

Reply:
no, 137 is in the constructor, the value might have been changed due to a response, however if a following response does not have a keep-alive hint it should go back.

Reply:
Ah, I get it now :D